### DISCUSSION TITLE
Date Type Column Not Included in attributes

### DISCUSSION THREAD

**Discussion by sethdumaguin** [visgl/deck.gl#5723] [General]
Hello!

I have an MVTLayer with a `last_updated` column that is a date type as one of its attributes. When I click on a geometry for that layer, it seems like `last_updated` is not included in the object.properties that gets passed in inside the `onClick`. Any ideas as to why it's not being included? Thanks

**Comment by padawannn** [comment]
According to the specification, in mvt format attributes are encoded as number or string so it sounds like a problem with the system that encodes your data in mvt format. Can you give us an example of the MVT that you are using?

**Reply by sethdumaguin** [reply]
I can describe to you what the data looks like. We have a `last_updated` column that's a `date type` and other columns that are string and number types. We're using carto maps API to fetch the mvt url for the MVTLayer to use. And it seems like the `last_updated` column is not included when we click on features.

**Reply by sethdumaguin** [reply]
For example: the dataset is

name | age | last_updated | the_geom
Foo    | 21   | 5-5-2021       | POINT(0 0)
Bar     | 22   | 5-2-2021       | POINT(1 1)

when we click on POINT(0 0) the onClick would only give us 

`
 {
    name: "Foo"
    age: 21
 }
`

**Reply by padawannn** [reply]
If you're using Carto Maps API it's probably a problem with your query. Have you tried to convert date data into a string in the query that you are sending to the Carto Maps API. For example

`SELECT name, age, to_char(last_updated, 'YYYY-MM-DD')  AS updated, the_geom, the_geom_webmercator FROM your_table `

**Reply by dgaubert** [reply]
Hi! 

MVT doesn't support every type in Postgres, the supported types are: 
- bool
- int8, int2, int4
- float4, float8
- numeric
- bpchar, varchar, text, literal

So, as @padawannn mentioned above, you might need to cast the required column to one type of the supported list in the query of the layer. You can cast dates as timestamp from epoch as follows:

```sql
select name, age, date_part('epoch', "last_updated") as "last_updated", ... from your_table
```

**Reply by ibgreen** [reply]
@padawannn Just a thought: it could potentially be helpful to add @dgaubert general comments to the @loaders.gl/mvt documentation, and reference those docs from the deck.gl MVTLayer

**Reply by Pessimistress** [reply]
@ibgreen we already have a link in the MVTLayer doc: https://deck.gl/docs/api-reference/geo-layers/mvt-layer#loadoptions

**Reply by ibgreen** [reply]
OK great, then adding a table with supported data types and a sample SQL stanza to the MVT loader docs would do it :)
